src/v/archival/tests/archival_metadata_stm_gtest.cc

```diff
@@ -232,8 +245,23 @@ TEST_F_CORO(
             cluster::segment_validated::yes);
       });
 
-    co_await std::move(delay_applied);
+    co_await sent_first_append.wait(10s);
+
+    // Expecting this to fail as we have the replication blocked.
+    auto sync_result_before_replication = co_await with_leader(
+      10s, [this, &plagued_node](raft::raft_node_instance& node) mutable {
+          if (node.get_vnode() != plagued_node) {
+              throw std::runtime_error{"Leadership moved"};
+          }
+
+          return get_leader_stm().sync(10ms);
```

Comment:
Asserts that sync fails in-between a replicate() call and replication actually happening. Can remove completely the other test too.